dist/src/main/java/io/camunda/zeebe/shared/AbstractConfigInfoContributor.java

```diff
@@ -0,0 +1,107 @@
+ */
+package io.camunda.zeebe.shared;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.datatype.jsr310.PackageVersion;
+import java.io.IOException;
+import java.util.Map;
+import org.springframework.boot.actuate.endpoint.SanitizableData;
+import org.springframework.boot.actuate.endpoint.SanitizingFunction;
+import org.springframework.boot.actuate.info.Info.Builder;
+import org.springframework.boot.actuate.info.InfoContributor;
+import org.springframework.util.unit.DataSize;
+
+public abstract class AbstractConfigInfoContributor implements InfoContributor {
+  private static final TypeReference<Map<String, Object>> TYPE_REFERENCE = new TypeReference<>() {};
+
+  private final Object config;
+  private final ObjectMapper mapper;
+
+  protected AbstractConfigInfoContributor(
+      final Object config, final SanitizingFunction sanitizingFunction) {
+    this.config = config;
+
+    final var sanitizingModule = new SanitizingModule(sanitizingFunction);
+    mapper = new ObjectMapper();
+    mapper
+        .registerModule(sanitizingModule)
+        .registerModule(new JavaTimeModule())
+        .disable(SerializationFeature.WRITE_DURATIONS_AS_TIMESTAMPS);
+  }
+
+  @Override
+  public void contribute(final Builder builder) {
+    final var sanitized = mapper.convertValue(config, TYPE_REFERENCE);
+    builder.withDetail("config", sanitized);
+  }
+
+  private static final class SanitizingModule extends SimpleModule {
+
+    private SanitizingModule(final SanitizingFunction sanitizingFunction) {
+      super(SanitizingModule.class.getName(), PackageVersion.VERSION);
+      addSerializer(DataSize.class, new DataSizeSerializer());
+      addSerializer(String.class, new StringSanitizer(sanitizingFunction));
+    }
+  }
+
+  private static final class DataSizeSerializer extends StdSerializer<DataSize> {
+
+    private DataSizeSerializer() {
+      super(DataSize.class);
+    }
+
+    @Override
+    public void serialize(
+        final DataSize dataSize,
+        final JsonGenerator jsonGenerator,
+        final SerializerProvider serializerProvider)
+        throws IOException {
+      if (dataSize.toMegabytes() > 0) {
+        jsonGenerator.writeString(dataSize.toMegabytes() + "MB");
+      } else if (dataSize.toKilobytes() > 0) {
+        jsonGenerator.writeString(dataSize.toKilobytes() + "KB");
+      } else {
+        jsonGenerator.writeString(dataSize.toBytes() + "B");
+      }
+    }
+  }
+
+  private static final class StringSanitizer extends StdSerializer<String> {
```

Comment:
Initially wanted to make it for all objects, but the problem is you end up with a recursive call - there's no way to say "serializeObjectWithoutUsingThisSerializer" :smile: